File: 0189-rotate-array/solution.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k=k%n;
        int r = n-k-1;
        if(n==1)
            return ;
        reverse(nums,0,r);
        reverse(nums,r+1,n-1);
        reverse(nums,0,n-1);
    }

    private void reverse(int[] nums , int l , int r)
    {
        while(l<r)
        {
            int temp = nums[l];
            nums[l] = nums[r];
            nums[r] = temp;
            l++;
            r--;
        }
    }
}

File: 0189-rotate-array/solution.c
void rotateArray(int* nums ,int x , int y){
    while (y > x){
        int temp = nums[x];
        nums[x] = nums[y];
        nums[y] = temp;
        x++;
        y--;        
    }

}

void rotate(int* nums, int numsSize, int k) {
   int t = k %numsSize;
   rotateArray(nums , 0 , numsSize-1); 
   rotateArray(nums , 0 , t-1); 
   rotateArray(nums , t , numsSize-1); 
}

